tools/cli/lib/api.c

```diff
@@ -948,13 +960,60 @@ TDNFCliCheckUpdateCommand(
     }
     else
     {
-        for(dwIndex = 0; dwIndex < dwCount; ++dwIndex)
+        // Format output like `tdnf list updates`, which is compatible with yum/dnf
+        if (nCheckUpdateCompat)
```

Comment:
This code is copied from `TDNFCliListCommand()`, correct?. Can you please put this into a function to avoid code duplication?

Reply:
There's actually a lot of duplicate code between those two functions, more than just that output. I'll take a look at what I can do.

Reply:
Added `TDNFCliListPackagesPrint` function (open to better naming suggestions) that is called by both `TDNFCliListCommand` and `TDNFCliCheckUpdateCommand` to do json and pretty table formatted output. Uses the old logic if nJsonOutput or nCheckUpdateCompat are not true.

From running the commands - with and without json - the output looks the same to me. And with nCheckUpdateCompat set to true, `list updates` and `check-update` also has the same output, with the additional exit code.